src/pages/documents/index.astro

```diff
@@ -3,12 +3,27 @@ import { getCollection } from "astro:content";
 
 import Layout from "@/layouts/base.astro";
 
-// Fetch all documents from the 'tags' collection
+const currentUrl = Astro.url.origin + "/RbxDocs";
 let documents = await getCollection("documents");
+
+function getThumbnail(link: string) {
+    const linkRegex = /https?:\/\/[^\s]+/;
+    const containsLink = link.match(linkRegex);
+
+    return containsLink ? containsLink[0] : `${currentUrl}/thumbnails/${link}`;
+}
```

Comment:
Convert to arrow function for consistency (unless it's an Astro function, then leave as is and ignore this comment).